proseco/acq.py

```diff
@@ -568,12 +568,49 @@ def get_initial_catalog(self):
 
         self.log(f'Getting initial catalog from {len(cand_acqs)} candidates')
 
-        # Start the lists of acq indices and box sizes with the values from
-        # the include lists.  Usually these will be empty.
-        acq_indices = [cand_acqs.get_id_idx(id) for id in self.include_ids]
-        box_sizes = self.include_halfws[:]  # make a copy
-
-        # Accumulate indices and box sizes of candidate acq stars that meet
+        # Build up the initial catalog as a list of indices into cand_acqs
+        # and the corresponding initial box size (halfw).
+        acq_indices = []
+        box_sizes = []
+
+        # Start with force-include stars, if any.
+        if self.include_ids:
+            self.log(f'Processing force-include ids={self.include_ids} '
+                     f'halfws={self.include_halfws}')
+
+            # Re-order candidate acqs to put those in the include list first
+            ok = np.in1d(cand_acqs['id'], self.include_ids)
+            idxs = np.concatenate([np.where(ok)[0], np.where(~ok)[0]])
+            cand_acqs = cand_acqs[idxs]
+
+            n_include = len(self.include_ids)
+            for min_p_acq in (0.75, 0.5, 0.25, 0.05, -1):
+                if len(acq_indices) < n_include:
+                    # Select candidates meeting min_p_acq, and update
+                    # acq_indices, box_sizes in place
+                    self.select_best_p_acqs(cand_acqs[:n_include], min_p_acq,
+                                            acq_indices, box_sizes)
+                else:
+                    # Because of the final min_p_acq = -1, this is guaranteed to finish.
```

Comment:
That comment may not have been the most helpful.  `select_best_p_acqs` is gradually adding cand_acq stars to `acq_indices` in place, allowing stars that meet a (progressively smaller) minimum value for acquisition probability `p_acq`.  Without the final `-1` in the `min_p_acq` loop, it would happen that an included star with best-case `p_acq < 0.05` would not be included.  In the loop down lower for normal stars, the last bin is 0.05, which means that `proseco` will never select an acq star with acquisition probability below 5%.  But for forced stars anything goes.

It is true the `else: break` could be removed at the cost of potentially doing a few unnecessary iterations.  In fact that is << 1 microsec, so I'll do that for code simplicity.